src/common/transformations/src/transformations/op_conversions/convert_gather_to_compressed.cpp

```diff
@@ -107,24 +107,34 @@ ov::pass::ConvertGatherToGatherCompressed::ConvertGatherToGatherCompressed() {
         std::shared_ptr<ov::Node> gather_input_zp = optional_zero_point;
         std::vector<std::shared_ptr<ov::Node>> result_nodes = {};
 
+        ov::element::Type output_type = gather_node->get_output_element_type(0);
+        // If Convert exists in scale branch, it means that the Scale Constant may be strored in other precision other
+        // than FP32. The Convert is inserted here to maintain graph correctness. GatherCompressed's output should
+        // follow the precision of the get_destination_type of Convert. The Convert should be kept, and later let the
+        // plugin decides whether it could optimize out this Convert
+        if (pattern_map.count(last_convert_m)) {
+            auto convert_node =
+                std::dynamic_pointer_cast<ov::op::v0::Convert>(pattern_map.at(last_convert_m).get_node_shared_ptr());
+            output_type = convert_node->get_destination_type();
+            gather_input_scale = pattern_map.at(last_convert_m).get_node_shared_ptr();
+        }
+
         std::shared_ptr<ov::Node> new_gather_node = nullptr;
         if (with_zero_point) {
-            new_gather_node =
-                std::make_shared<ov::op::internal::GatherCompressed>(gather_input_a,
-                                                                     gather_input_b,
-                                                                     gather_input_c,
-                                                                     gather_node->get_batch_dims(),
-                                                                     gather_input_scale,
-                                                                     gather_input_zp,
-                                                                     gather_node->get_output_element_type(0));
+            new_gather_node = std::make_shared<ov::op::internal::GatherCompressed>(gather_input_a,
+                                                                                   gather_input_b,
+                                                                                   gather_input_c,
+                                                                                   gather_node->get_batch_dims(),
+                                                                                   gather_input_scale,
+                                                                                   gather_input_zp,
+                                                                                   output_type);
```

Comment:
BTW do you know why we implicitly set `output_type`? Is it needed to avoid precision changes during `ConvertPrecision` pass? If yes, why must we store original precision of `GatherCompressed`?

Reply:
Good question, this is a leftover from @xipingyan, do you think we can remove this precision setting in class construction ?

Reply:
Remove `output_type` from constructor